tests/cpp/lp_namer/ActiveLinkTest.cpp

```diff
@@ -2,10 +2,209 @@
 #include "ActiveLinks.h"
 
 
-TEST(ActiveLinkTests, empty_test)
+TEST(ActiveLinkTests, valid_candidate_data)
 {
-    auto active_links = ActiveLinks();
-    ASSERT_EQ(1, 1);
+    CandidateData cand1;
+    cand1.link_id =1;
+    cand1.name = "transmission_line_1";
+    cand1.link = "area1 - area2";
+    cand1.linkor = "area1";
+    cand1.linkex = "area2";
+    
+    std::vector<CandidateData> cand_data_list;
+    cand_data_list.push_back(cand1);
+
+    std::map<std::string, LinkProfile> profile_map;
+
+    ActiveLinks links;
+
+    links.addCandidates(cand_data_list, profile_map);
+
+    ASSERT_EQ(links.size(), 1);
+
+}
+
+TEST(ActiveLinkTests, multiple_valid_candidates_data_on_same_link)
+{
+    CandidateData cand1;
+    cand1.link_id = 1;
+    cand1.name = "transmission_line_1";
+    cand1.link = "area1 - area2";
+    cand1.linkor = "area1";
+    cand1.linkex = "area2";
+
+    CandidateData cand2;
+    cand2.link_id = 1;
+    cand2.name = "transmission_line_2";
+    cand2.link = "area1 - area2";
+    cand2.linkor = "area1";
+    cand2.linkex = "area2";
+
+    std::vector<CandidateData> cand_data_list;
+    cand_data_list.push_back(cand1);
+    cand_data_list.push_back(cand2);
+
+    std::map<std::string, LinkProfile> profile_map;
+
+    ActiveLinks links;
+
+    links.addCandidates(cand_data_list, profile_map);
+
+    ASSERT_EQ(links.size(), 1);
 
 }
 
+TEST(ActiveLinkTests, multiple_valid_candidates_data_on_different_link)
+{
+    CandidateData cand1;
+    cand1.link_id = 1;
+    cand1.name = "transmission_line_1";
+    cand1.link = "area1 - area2";
+    cand1.linkor = "area1";
+    cand1.linkex = "area2";
+
+    CandidateData cand2;
+    cand2.link_id = 2;
+    cand2.name = "pv";
+    cand2.link = "area1 - pv";
+    cand2.linkor = "area1";
+    cand2.linkex = "pv";
+
+    std::vector<CandidateData> cand_data_list;
+    cand_data_list.push_back(cand1);
+    cand_data_list.push_back(cand2);
+
+    std::map<std::string, LinkProfile> profile_map;
+
+    ActiveLinks links;
+
+    links.addCandidates(cand_data_list, profile_map);
+
+    ASSERT_EQ(links.size(), 2);
+
+}
+
+TEST(ActiveLinkTests, multiple_candidate_same_name_on_same_link)
+{
+    CandidateData cand1;
+    cand1.link_id = 1;
+    cand1.name = "transmission_line_1";
+    cand1.link = "area1 - area2";
+    cand1.linkor = "area1";
+    cand1.linkex = "area2";
+
+    CandidateData cand2;
+    cand2.link_id = 1;
+    cand2.name = "transmission_line_1";
+    cand2.link = "area1 - area2";
+    cand2.linkor = "area1";
+    cand2.linkex = "area2";
+
+    std::vector<CandidateData> cand_data_list;
+    cand_data_list.push_back(cand1);
+    cand_data_list.push_back(cand2);
+
+    std::map<std::string, LinkProfile> profile_map;
+
+    try {
+        ActiveLinks links;
+        links.addCandidates(cand_data_list, profile_map);
+        FAIL() << "duplicate not detected";
+    }
```

Comment:
if we modify the test like this it should pass but it doesn't.

```suggestion
  std::vector<CandidateData> cand_data_list;
    cand_data_list.push_back(cand1);
    std::vector<CandidateData> cand_data_list_2;
    cand_data_list_2.push_back(cand2);

    std::map<std::string, LinkProfile> profile_map;
    
    ActiveLinks links;
    links.addCandidates(cand_data_list, profile_map);
    try {
        links.addCandidates(cand_data_list_2, profile_map);
        FAIL() << "duplicate not detected";
    }
```

The check on the presence of a candidate in the `ActiveLinks` is done only once on the list of `candidate_data` passed as argument.
If we call `add_candidates` twice with two vectors of candidates data the check are not properly performed .

I propose a change of the `ActiveLinks` class,, we can discuss it later